firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/LiveGenerativeModel.kt

```diff
@@ -0,0 +1,122 @@
+import io.ktor.client.plugins.websocket.webSocketSession
+import io.ktor.websocket.Frame
+import io.ktor.websocket.close
+import io.ktor.websocket.readBytes
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * Represents a multimodal model (like Gemini) capable of real-time content generation based on
+ * various input types, supporting bidirectional streaming.
+ */
+public class LiveGenerativeModel
+internal constructor(
+  private val modelName: String,
+  private val config: LiveGenerationConfig? = null,
+  private val tools: List<Tool>? = null,
+  private val systemInstruction: Content? = null,
+  private val location: String,
+  private val controller: APIController
+) {
+  internal constructor(
+    modelName: String,
+    apiKey: String,
+    firebaseApp: FirebaseApp,
+    config: LiveGenerationConfig? = null,
+    tools: List<Tool>? = null,
+    systemInstruction: Content? = null,
+    location: String = "us-central1",
+    requestOptions: RequestOptions = RequestOptions(),
+    appCheckTokenProvider: InteropAppCheckTokenProvider? = null,
+    internalAuthProvider: InternalAuthProvider? = null,
+  ) : this(
+    modelName,
+    config,
+    tools,
+    systemInstruction,
+    location,
+    APIController(
+      apiKey,
+      modelName,
+      requestOptions,
+      "gl-kotlin/${KotlinVersion.CURRENT} fire/${BuildConfig.VERSION_NAME}",
+      firebaseApp,
+      AppCheckHeaderProvider(TAG, appCheckTokenProvider, internalAuthProvider),
+    ),
+  )
+
+  /**
+   * Returns a LiveSession object using which you could send/receive messages from the server
+   * @return LiveSession object created. Returns null if the object cannot be created.
+   * @throws [BidiServerHandshakeFailed] if the handshake with the server failed.
+   */
+  public suspend fun connect(): LiveSession {
+    val client = HttpClient(OkHttp) { install(WebSockets) }
+
+    val bidiEndPoint = this.controller.getBidiEndpoint(location)
+    val clientMessage =
+      BidiGenerateContentClientMessage(
+          this.modelName,
+          this.config?.toInternal(),
+          this.tools?.map { it.toInternal() },
+          this.systemInstruction?.toInternal()
+        )
+        .toInternal()
+    val data: String = Json.encodeToString(clientMessage)
+    val webSession = client.webSocketSession(bidiEndPoint)
+    webSession.send(Frame.Text(data))
+    var shouldReturn = false
+    webSession.let {
+      val serverMessage = it.incoming.receive()
+      val receivedBytes = (serverMessage as Frame.Binary).readBytes()
+      val receivedJson = receivedBytes.toString(Charsets.UTF_8)
+      // TODO: Try to decode the json instead of string matching.
+      if ("setupComplete" in receivedJson) {
+        shouldReturn = true
+      }
+    }
+    return if (shouldReturn) {
+      LiveSession(session = webSession, isRecording = false)
+    } else {
+      webSession.close()
+      throw BidiServerHandshakeFailed()
+    }
```

Comment:
No need for the `let` block

```suggestion
    val receivedJson = webSession.incoming.receive().readBytes().toString(Charsets.UTF_8)
    // TODO: Try to decode the json instead of string matching.
    return if("setupComplete" in receivedJson) {
      LiveSession(session = webSession, isRecording = false)
    } else {
      webSession.close()
      throw BidiServerHandshakeFailed()
    }
```